src/Components/Facility/Consultations/DailyRoundsFilter.tsx

```diff
@@ -0,0 +1,130 @@
+import { Popover, Transition } from "@headlessui/react";
+import ButtonV2 from "../../Common/components/ButtonV2";
+import { Fragment } from "react";
+import { SelectFormField } from "../../Form/FormFields/SelectFormField";
+import TextFormField from "../../Form/FormFields/TextFormField";
+import CareIcon from "../../../CAREUI/icons/CareIcon";
+import dayjs from "dayjs";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { DailyRoundsFilterModel } from "../models";
+
+export const DailyRoundsFilter = (props: {
+  onApply: (filter: any) => void;
+}) => {
+  const [filter, setFilter] = useState<DailyRoundsFilterModel>({
+    rounds_type: undefined,
+    fromDateTime: undefined,
+    toDateTime: undefined,
+  });
+
+  const clearFilter = () => {
+    const clearedFilter = {
+      rounds_type: undefined,
+      fromDateTime: undefined,
+      toDateTime: undefined,
+    };
+
+    setFilter(clearedFilter);
+    props.onApply({ query: filter });
```

Comment:
`filter` will still be the old value, because the component has no rerendered.

Reply:
@rithviknishad ,
 what should be present in use effect hook dependency
 Irrespective of clearFilter even onClick applyFilter is also not working

Reply:
I'm looking into it